### PR TITLE
Update Android and Linux-Bionic images

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Also update the OpenSSL hack based on the new setup.

Contributes to https://github.com/dotnet/runtime/issues/102851 (fixes it for our docker images, but not in general).

### CODE REVIEW

**Review by sbomer — APPROVED** [review]

File: src/native/libs/System.Security.Cryptography.Native/extra_libs.cmake [hunk 1]
```diff
@@ -4,26 +4,27 @@ macro(append_extra_cryptography_libs NativeLibsExtra)
...
+    find_package(OpenSSL)
     # This is bad and wrong, but good enough to satisfy the build
     # We only care about having "enough" OpenSSL to get the native lib built
     # here, and it's on the end user to ship libssl/libcrypto from Google
-    if(FORCE_ANDROID_OPENSSL)
-        set(OPENSSL_CRYPTO_LIBRARY /usr/lib/x86_64-linux-gnu/libcrypto.so)
-        set(OPENSSL_SSL_LIBRARY /usr/lib/x86_64-linux-gnu/libssl.so)
+    if(FORCE_ANDROID_OPENSSL AND NOT OPENSSL_FOUND)
+        set(OPENSSL_CRYPTO_LIBRARY /usr/lib/x86_64-linux-gnu/libcrypto.so CACHE PATH "libcrypto.so" FORCE)
+        set(OPENSSL_SSL_LIBRARY /usr/lib/x86_64-linux-gnu/libssl.so CACHE PATH "libcrypto.so" FORCE)
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply src/native/libs/System.Security.Cryptography.Native/extra_libs.cmake:13]: There's no `FILEPATH` option here, `PATH` is the correct CMake type.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**